addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/HyperionDiscoveryParticipant.java

```diff
@@ -58,6 +58,10 @@ public DiscoveryResult createResult(ServiceInfo service) {
 
         String longName = service.getName();
         int pos = longName.indexOf("@");
+        if (pos < 0 || pos >= longName.length()) {
```

Comment:
I know, that was more of a guard in case someone in the future changes the indexOf() call to something else which may not always return a valid index position.  

I thought you had found something more sinister :)